### PR TITLE
Milestone 6: Friendships V2

### PR DESCRIPTION
**Pull request by AndresFMoya**
Good day TSEs.

Friendships with 2 rows for mutual friendship were implemented. Looking to hear your comments.

Best!

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1,7 +1,3 @@
 <div>
-  <% @users.each do |user| %>
-    <ul>
-      <%= link_to user.name, user %>
-    </ul>
-  <% end %>
+ <%= render @users %>
```
Comment (adriaanbd) [inline app/views/users/index.html.erb]: This list of users should not include the `current_user`. At the moment, it does! [`OPTIONAL`]

**Review by adriaanbd — CHANGES_REQUESTED** [review]
Please guarantee the uniqueness of user and friend relationships in regards to both a confirmed and unconfirmed Friendship [`REQUIRED`].

Two unconfirmed friendships from the same relationship (user_id 1 / friend_id 3 and user_id 3 / friend)id 1):

```
irb(main):002:0> Friendship.all
=> #<Friendship id: 2, user_id: 1, friend_id: 3, confirmed: nil, created_at: "2019-11-29 06:05:48", updated_at: "2019-11-29 06:05:48">]>

irb(main):003:0> f3 = Friendship.create!(user: User.third, friend: User.first)
   (20.3ms)  COMMIT
=> #<Friendship id: 3, user_id: 3, friend_id: 1, confirmed: nil, created_at: "2019-11-29 06:15:15", updated_at: "2019-11-29 06:15:15">
```

Two confirmed friendships from the same relationship (`f3` operation should rollback and/or fail validation):

```
irb(main):008:0> f2
=> #<Friendship id: 2, user_id: 1, friend_id: 3, confirmed: true, created_at: "2019-11-29 06:05:48", updated_at: "2019-11-29 06:29:42">
irb(main):009:0> f3.update(confirmed: true)
   (0.4ms)  BEGIN
  Friendship Update (0.8ms)  UPDATE "friendships" SET "confirmed" = $1, "updated_at" = $2 WHERE "friendships"."id" = $3  [["confirmed", true], ["updated_at", "2019-11-29 06:30:04.642374"], ["id", 3]]
   (32.5ms)  COMMIT
=> true
```

Likewise, a user shouldn't be able to friend themselves:

```
irb(main):010:0> f4 = Friendship.create!(user: User.third, friend: User.third)
   (34.5ms)  COMMIT
=> #<Friendship id: 4, user_id: 3, friend_id: 3, confirmed: nil, created_at: "2019-11-29 06:32:18", updated_at: "2019-11-29 06:32:18">
```

Please fix the above-mentioned issues and request another code review.

Status: `Changes Requested`
Reviewer: `Adriaan Beiertz`

Feel free to contact me via email, Slack or initiate discussion herein.

**Comment by KerronKing** [comment]
Good day @adriaanbd. Thanks for your previous feedback. The requested changes were made.

Cheers!

**Review by adriaanbd — CHANGES_REQUESTED** [review]
Great work on guaranteeing uniqueness on a confirmed friendship relationship. However, we need to do the same for when `confirmed: nil` as well. In other words, if there is a pending friend request from user 1 to user 3, there can't be another friend request from user 3 to user 1.

Take a look:

```
irb(main):003:0> Friendship.all
=>[#<Friendship id: 1, user_id: 1, friend_id: 2, confirmed: true, created_at: "2019-11-29 22:48:25", updated_at: "2019-11-29 22:48:25">, 

#<Friendship id: 2, user_id: 1, friend_id: 3, confirmed: nil, created_at: "2019-11-29 22:48:25", updated_at: "2019-11-29 22:48:25">,

 #<Friendship id: 3, user_id: 3, friend_id: 1, confirmed: nil, created_at: "2019-11-29 22:49:55", updated_at: "2019-11-29 22:49:55">]>
```

Friendship.third shouldn't have been committed to the database; it should've rolled back like the other one. 

Likewise, good work on preventing a user to have a friendship with itself.

Please correct this and submit another code review request.

Almost ready to :ship: 

Status: `Changes Requested`
Reviewer: `Adriaan Beiertz`

**Comment by AndresFMoya** [comment]
Good day @adriaanbd. Thanks for pointing us on the right path. We add the pending validation to prevent a user from sending a friend request to another user who previously sent one. Looking to hear your comments.

Best!

**Review by adriaanbd — APPROVED** [review]
Okay we're ready to ship, thanks for incorporating the requested changes. 

By the way, `current_user` is still displaying in the list of users obtained from visiting the Friend Requests link. I'd suggest for you to remove `current_user` from this list because it doesn't make sense for it to be there since you can't request a friendship with yourself. Nevertheless, this is an optional requirement.

Status: `Approved`
Reviewer: `Adriaan Beiertz`

**Pull request merged**